src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -200,7 +213,7 @@ internal async Task InnerProcessRequestsAsync<TContext>(IHttpApplication<TContex
             {
                 while (true)
                 {
-                    var streamContext = await _multiplexedContext.AcceptAsync();
+                    var streamContext = await _multiplexedContext.AcceptAsync(_connectionAborted.Token);
```

Comment:
I took a look at `MultiplexedConnectionContext`.

What is preferred between cancelling with a cancellation token, vs directly calling `Abort` on the context? I'm guessing that aborting the connection would automatically cause AcceptAsync to return or throw.

Is there overhead involved with passing in the token with each call to AcceptAsync?

Reply:
> Is there overhead involved with passing in the token with each call to AcceptAsync?

That's a good point. Generally speaking, any time you pass a cancellable CancellationToken to an async method that cannot complete immediately it will require an extra allocation for the CancellationToken registration.

In this case, [MsQuicListener.AcceptAsync](https://github.com/dotnet/aspnetcore/blob/9402ebf905c7896c8f2fda77ea3e702d692fc37a/src/Shared/runtime/Quic/Implementations/MsQuic/MsQuicListener.cs#L72) is using a bounded Channel that has to take the [slow path and allocate a new AsyncOperation](https://github.com/dotnet/runtime/blob/7939172ea4d0052901215e83fb59e7172dcd13d0/src/libraries/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs#L137-L159) for each accepted connection.

Reply:
I could try calling Abort instead of cancelling a token; I'll see how that plays out.